src/web/error.rs

```diff
@@ -80,3 +112,71 @@ impl From<ParseError> for ApiError {
         HawkErrorKind::Parse(SyncFailure::new(inner)).into()
     }
 }
+
+impl From<ValidationErrorKind> for ApiError {
+    fn from(kind: ValidationErrorKind) -> Self {
+        let validation_error: ValidationError = Context::new(kind).into();
+        validation_error.into()
+    }
+}
+
+impl Serialize for ValidationError {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        Serialize::serialize(&self.inner.get_context(), serializer)
+    }
+}
+
+impl Serialize for ValidationErrorKind {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut seq = serializer.serialize_seq(None)?;
+
+        match *self {
+            ValidationErrorKind::FromDetails(ref description, ref location, ref name) => {
+                seq.serialize_element(&SerializedValidationError {
+                    description,
+                    location,
+                    name: name.as_ref().map(|name| &**name),
+                    value: None,
+                })?;
+            }
+
+            ValidationErrorKind::FromValidationErrors(ref errors, ref location) => {
+                errors.clone().field_errors().iter().fold(
```

Comment:
Oh, almost forgot! This `clone()` is because of a borrowck error I was getting.

`errors.field_errors()` takes `self`, but I want to call it on a reference to `errors` so was getting the dreaded "cannot move out of borrowed content". Doing the extra `clone()` feels like it should be unnecessary though, is there something else I should be doing?

Reply:
ref https://github.com/Keats/validator/issues/65

Reply:
I think this block would be improved by just a couple for loops. Since we're not really accumulating anything via fold (if I'm reading it right, we just use the accumulator to break out of the fold after an error)

```
                for (field, field_errors) in errors.clone().field_errors() {
                    for field_error in field_errors {
                        seq.serialize_element(&SerializedValidationError {
                            description: &field_error.code,
                            location,
                            name: Some(field),
                            value: field_error.params.get("value"),
                        })?;
                    }
                }
```

(I think that's equivalent?)